standard-controller/src/main/java/io/enmasse/controller/standard/StandardController.java

```diff
@@ -100,9 +102,23 @@ public void start() throws Exception {
                 .withUid(addressSpace.getMetadata().getUid())
                 .build();
 
+        ConfigMapAddressApi addressApi = new ConfigMapAddressApi(kubeClient, options.getInfraUuid(), ownerReference);
+
+        // Replace resources as part of upgrade if version is different
+        for (Address address : addressApi.listAddresses(options.getAddressSpaceNamespace())) {
+            if (!options.getVersion().equals(address.getAnnotation(AnnotationKeys.VERSION))) {
```

Comment:
What makes you drive this from version rather than checking for the expected owner reference (address.getMetadata().getOwnerReferences().contains(ownerReference))?

If driving from version is the way to go, doesn't io.enmasse.k8s.api.ConfigMapAddressApi#create need to set the version too?

Reply:
@k-wall The ownerreference is set on the configmap and not on the Address object. This is because finalizers do not work with our custom resources at this point. So checking became a bit clunky, and I thought an annotation on the address would perhaps be desireable anyway.

You are right it should be set on the address create as well!.

I am also working on adding support to do the conversion in the agent as well (for the brokered address space).